pennylane_lightning/lightning_qubit.py

```diff
@@ -131,6 +145,62 @@ def apply_lightning(self, state, operations):
 
         return np.reshape(state_vector, state.shape)
 
+    def adjoint_jacobian(self, tape, starting_state=None, use_device_state=False):
+        if not CPP_BINARY_AVAILABLE:
+            return super().adjoint_jacobian(tape, starting_state, use_device_state)
+
+        if self.shots is not None:
+            warn(
+                "Requested adjoint differentiation to be computed with finite shots."
+                " The derivative is always exact when using the adjoint differentiation method.",
+                UserWarning,
+            )
```

Comment:
It would be great it we could add support for sampling from the inner-products computed for the adjoint method 💪

Reply:
Added as a story.

Reply:
@josh146, what do you mean by this? Is this following the idea of supporting adjoint in a finite-shots setting?

Reply:
Yep - the final inner product that is performed as part of the adjoint method, rather than computing that inner product exactly, could we sample from it using the number of shots the user has requested?

This is partly a research question though, since I don't think the answer here is straightforward 🤔 It would give an approximation of 'parameter shift + finite shots', but I don't know if it's a reasonable approximation or not

Reply:
Thanks! Yes for me this is also quite an open question. Since the adjoint method itself depends on evolving the _state_ and finding overlaps, I'm not sure how this would be hardware compatible.

Reply:
> I'm not sure how this would be hardware compatible.

Oh it definitely wouldn't be, but I was thinking more maybe we could _approximate_ the stochasticity of finite-shots with the adjoint method via some approach?

Reply:
Basically: even if we can't sample directly from this (unphysical) expectation value, can we treat the exact inner product as the mean of some probability distribution, with variance dependent on shot value?

Reply:
I think this is something we can explore. I've created a story for this, so we can revisit it and figure out the best path.

Reply:
Nice! I had an idea; if we can work out the adjoint rule for differentiating probabilities, and then write all expval derivatives in terms of this (via `expval = \sum_i \lambda_i probs_i`), finite-shots would make sense and probably 'just work', since we have a probability distribution we can sample from!

Although, there is a partial downside --- I think this might scale (linearly) with the number of qubits. This is because reverse-mode differentiation scales as per the size of the output, unlike forward-mode differentiating (e.g., parameter-shift), which scales with the size of the input.